### PR TITLE
Add optional id to GangaDiracError

### LINKED ISSUE #897: DiracError needs to associate to jobID
DiracError should be a bit more helpful and possibly take an optional jobID or filename to make debugging a DIRAC error (sometimes a little too generic) easier.

### PR DESCRIPTION
**Pull request by mesmith75**
Fixes #897 . This adds an optional id to the GangaDiracError. Not sure how useful this really is as there doesn't seem any obvious place at the moment where it is needed.

Also removed unused import in LHCbDataset.

### CODE REVIEW

File: python/GangaDirac/Lib/Utilities/DiracUtilities.py [hunk 1]
```diff
@@ -26,6 +26,14 @@
 
 class GangaDiracError(GangaException):
     """ Exception type which is thrown from problems executing a command against DIRAC """
+    def __init__(self, message, job_id = None):
```
Comment (rob-c) [inline python/GangaDirac/Lib/Utilities/DiracUtilities.py]: Is it worth adding the Ganga ID as well?

File: python/GangaDirac/Lib/Utilities/DiracUtilities.py [hunk 2]
```diff
@@ -26,6 +26,15 @@
 
 class GangaDiracError(GangaException):
     """ Exception type which is thrown from problems executing a command against DIRAC """
+    def __init__(self, message, dirac_id = None, job_id = None):
+        self.dirac_id = dirac_id
+        self.job_id = job_id
+        self.message = message
```
Comment (drmarkwslater) [inline python/GangaDirac/Lib/Utilities/DiracUtilities.py]: I'd suggest using the parent where possible, so maybe:
```
GangaException.__init__(self, message)
```

File: python/GangaDirac/Lib/Utilities/DiracUtilities.py [hunk 3]
```diff
@@ -26,6 +26,15 @@
...
+    def __str__(self):
+        if self.job_id and self.dirac_id:
+            return "GangaDiracError, Job %s with Dirac ID %s : %s" % (self.job_id, self.dirac_id, self.message)
+        else:
+            return "GangaDiracError: %s" % (self.message)
```
Comment (drmarkwslater) [inline python/GangaDirac/Lib/Utilities/DiracUtilities.py]: Similarly here:
```
else:
    return super(GangaException, self).__str__()
```

File: python/GangaDirac/Lib/Utilities/DiracUtilities.py [hunk 4]
```diff
@@ -26,6 +26,16 @@
 
 class GangaDiracError(GangaException):
     """ Exception type which is thrown from problems executing a command against DIRAC """
+    def __init__(self, message, dirac_id = None, job_id = None):
+        GangaException.__init__(self, message)
+        self.dirac_id = dirac_id
+        self.job_id = job_id
+        self.message = message
```
Comment (drmarkwslater) [inline python/GangaDirac/Lib/Utilities/DiracUtilities.py]: This isn't needed as it's handled in the base class 😄

**Pull request merged**